src/drag-scroll.ts

```diff
@@ -0,0 +1,575 @@
+    }
+  }
+
+  private setScrollBar(): void {
+    if (this.scrollbarHidden) {
+      this.hideScrollbar();
+    } else {
+      this.showScrollbar();
+    }
+  }
+
+  private getScrollbarWidth(): number {
+    // Return 0 is platform is server.
+    if (isPlatformServer(this._platformId)) {
+      return 0;
+    }
+
+
+
+    /**
+     * Browser Scrollbar Widths (2016)
+     * OSX (Chrome, Safari, Firefox) - 15px
+     * Windows XP (IE7, Chrome, Firefox) - 17px
+     * Windows 7 (IE10, IE11, Chrome, Firefox) - 17px
+     * Windows 8.1 (IE11, Chrome, Firefox) - 17px
+     * Windows 10 (IE11, Chrome, Firefox) - 17px
+     * Windows 10 (Edge 12/13) - 12px
+     */
+
+
+    const outer = this._renderer.createElement('div');
+    this._renderer.setStyle(outer, 'visibility', 'hidden');
+    this._renderer.setStyle(outer, 'width', '100px');
+    this._renderer.setStyle(outer, 'msOverflowStyle', 'scrollbar');  // needed for WinJS apps
+
+    // document.body.appendChild(outer);
+    this._renderer.appendChild(document.body, outer);
+    // this._renderer.appendChild(this._renderer.selectRootElement('body'), outer);
+
+    const widthNoScroll = outer.offsetWidth;
+    // force scrollbars
+    this._renderer.setStyle(outer, 'overflow', 'scroll');
+
+    // add innerdiv
+    const inner = this._renderer.createElement('div');
+    this._renderer.setStyle(inner, 'width', '100%');
+    this._renderer.appendChild(outer, inner);
+
+    const widthWithScroll = inner.offsetWidth;
+
+    // remove divs
+    this._renderer.removeChild(document.body, outer);
+
+    /**
+     * Scrollbar width will be 0 on Mac OS with the
+     * default "Only show scrollbars when scrolling" setting (Yosemite and up).
+     * setting default width to 20;
+     */
+    return widthNoScroll - widthWithScroll || 20;
+  }
+
+
+  /*
+   * Nav button
+   */
+  moveLeft() {
+    const childrenArray = this.snaps['_results'];
+    const contentElement = this._contentRef.nativeElement;
+    if (this.currIndex !== 0 && this.snaps.length !== 0) {
+      // reach left most
+      this.currIndex--;
+      clearTimeout(this.scrollToTimer);
+      this.scrollTo(contentElement, this.toChildrenLocation(), 500);
+    } else if (this.snaps.length === 0) {
+      this.currIndex--;
+      clearTimeout(this.scrollToTimer);
+      this.scrollTo(contentElement, this._contentRef.nativeElement.scrollLeft - 200, 500);
+    }
+  }
+
+  moveRight() {
+    const childrenArray = this.snaps['_results'];
+    const contentElement = this._contentRef.nativeElement;
+    if (!this.scrollReachesRightEnd && this.snaps.length !== 0 && childrenArray[this.currIndex + 1]) {
+      this.currIndex++;
+      clearTimeout(this.scrollToTimer);
+      this.scrollTo(contentElement, this.toChildrenLocation(), 500);
+    } else if (this.snaps.length === 0) {
+      this.currIndex++;
+      clearTimeout(this.scrollToTimer);
+      this.scrollTo(contentElement, this._contentRef.nativeElement.scrollLeft + 200, 500);
```

Comment:
Great job on making the navigation buttons working with no snaps. I wonder what if it is possible to move to the position of the next item instead of a fixed position.

Reply:
Yes, I was going to do that after you confirmed that it was fine. I'll see what the best approach is for getting the children without using `nativeElement`. I think I have a way to do that, will need to test.